public/js/saveApplicant.js

```diff
@@ -1,27 +1,62 @@
 document.getElementById('submitApplicant').addEventListener('click', e => {
-    e.preventDefault()
-    let data = getCompletedFormData()
-    let validate = validateFormInputs(data)
-    if (validate) {
-        makeApiRequest(data)
+    e.preventDefault();
+    let data = getCompletedFormData();
+    let validate = validateFormInputs(data);
+    let formIsValid = true;
+    Object.keys(validate).forEach(formItem => {
+        let itemKey = formItem;
+        let querySelector = `#${itemKey}Error`;
+        let formItemValues = validate[formItem];
+        Object.keys(formItemValues).forEach(validationType => {
+            let isValid = formItemValues[validationType];
+            if (!isValid) {
+                document.querySelector(querySelector).classList.add('alert-danger');
+                document.querySelector(querySelector).classList.remove('hidden');
+                document.querySelector(querySelector).innerHTML = errorMessage(validationType);
+                formIsValid = false;
+            } else {
+                document.querySelector(querySelector).classList.remove('alert-danger');
+                document.querySelector(querySelector).classList.add('hidden');
+                document.querySelector(querySelector).innerHTML = '';
+            }
+        })
+    });
+
+    if (formIsValid) {
+        makeApiRequest(data);
     } else {
-        document.getElementById('errorMsg').innerHTML = 'Not all fields have been filled out. Try again...'
+        document.getElementById('generalError').innerHTML = 'This form is invalid, please check all fields';
+        document.querySelector('#generalError').classList.remove('hidden');
```

Comment:
You are targetting the id generalError twice here yet using querySelector and getElementbyId. lets be consistent.

Reply:
yeah based off the old code and must have missed it - cheers!